source/agora/consensus/EnrollmentPool.d

```diff
@@ -169,19 +175,25 @@ public class EnrollmentPool
 
         Params:
             enroll_hash = key for an enrollment data which is hash of frozen UTXO
+            avail_height = height at which the enrollment is available
 
         Returns:
             true if the validator set has the enrollment data
 
     ***************************************************************************/
 
-    public bool hasEnrollment (const ref Hash enroll_hash) @trusted nothrow
+    public bool hasEnrollment (const ref Hash enroll_hash, Height avail_height)
+        @trusted nothrow
     {
         try
         {
-            auto results = this.db.execute("SELECT EXISTS(SELECT 1 FROM " ~
-                "enrollment_pool WHERE key = ?)", enroll_hash.toString());
-            return results.front().peek!bool(0);
+            auto results = this.db.execute("SELECT avail_height FROM " ~
+                "enrollment_pool WHERE key = ?", enroll_hash.toString());
+            if (!results.empty &&
+                Height(results.oneValue!(size_t)) == avail_height)
+                return true;
+            else
+                return false;
```

Comment:
Why would you write:
```D
if (condition)
    return true;
else
    return false;
```

Instead of simply `return condition;` ? I feel that this pattern is repeated way too frequently, please remember the [KISS principle](https://en.wikipedia.org/wiki/KISS_principle).